packages/venia-concept/src/components/ProductFullDetail/ProductFullDetail.js

```diff
@@ -135,11 +135,51 @@ class ProductFullDetail extends Component {
         );
     }
 
+    get mediaGalleryEntries() {
+        const { props, state } = this;
+        const { product } = props;
+        const { optionCodes, optionSelections } = state;
+        const {
+            configurable_options,
+            media_gallery_entries,
+            variants
+        } = product;
+        const isConfigurable = Array.isArray(configurable_options);
+
+        if (
+            !isConfigurable ||
+            (isConfigurable && optionSelections.size === 0)
+        ) {
+            return media_gallery_entries;
+        }
+
+        const item = variants.find(({ product: variant }) => {
+            for (const [id, value] of optionSelections) {
+                const code = optionCodes.get(id);
+
+                if (variant[code] !== value) {
+                    return false;
+                }
+            }
+
+            return true;
+        });
```

Comment:
This code, for identifying the selected variant, appears to be duplicated from `appendOptionsToPayload`, which is due to change in #1005 to accommodate custom attributes.

Reply:
Dang, nice catch - thanks <3

Reply:
I'm working on exporting this function so that both places can use it.